### ISSUE TITLE
Assignability check skipped for unique symbols

### ISSUE THREAD

**Issue by rbuckton** [Microsoft/TypeScript#21584]
**TypeScript Version:**  2.8.0-dev.20180202

In #20898 @rozzzly discovered a bug that was tangential to that issue:

> On a side note, I think I may have found a bug. So while typing up this issue, I found that I am able to achieve the desired behavior by using an intersection with a "tagged" object literal:
> ```ts
> export type FOO = symbol & { FOO: true };
> export const FOO: FOO = Symbol('FOO'); // ❓️ no error
> ```

We also don't see the error for declarations like:

```ts
const FOO: string = Symbol();
```

**Expected behavior:**

This should report the error: `error TS2322: Type 'unique symbol' is not assignable to type 'symbol & { FOO: true; }'.`

**Actual behavior:**

No error is reported. This is because `getESSymbolLikeTypeForNode` uses the `type` property on `SymbolLinks` which already has a type from the declaration, so it gets `symbol & { FOO: true }`.

**Related Issues:**

- #20898
